pdns/recursordist/docs/lua-scripting/functions.rst

```diff
@@ -21,3 +21,46 @@ These are some functions that don't really have a place in one of the other cate
   Get a random number.
 
   :param int upper_bound: The upper bound. You will get a random number below this upper bound.
+
+.. function:: spawnThread(script)
+
+   .. versionadded:: 5.2.0
+
+   Spawn a thread.
+
+   :param str script: The pathname of the Lua script to run.
+
+.. note::
+     The :func:`putIntoRecordCache` and :func:`getRecordCacheRecords` functions are experimental, their functionality might change in upcoming releases.
+
+.. function:: putIntoRecordCache(dump) -> int
+
+   .. versionadded:: 5.2.0
+
+   Put a record cache dump into the record cache.
+
+   :param str dump: The data in the proprietary format produced by :func:`getRecordCacheRecords`).
+   :returns: The number of records inserted into the record cache.
+
+   Some record sets might be skipped, for example when they are already present in the record cache or contain specific information not supported yet by this function.
+   If the :program:`Recursor` determines the version of the data is not compatible, it will skip loading and log an error.
```

Comment:
What is then returned? Looking at the code it's not clear to me if the exception is actually handled.

Reply:
An exception is handled at the bottom of `putRecordSets`. For the rest, see comment of latest commit 885268cac5d76cce60847cacac87380f9fbf651e